### PR TITLE
Validate language version opt-ins

### PR DESCRIPTION
**Pull request by sigurdm**
Fix of https://github.com/dart-lang/pub/issues/2316

This relies on unpublished changes in the analyzer.

### CODE REVIEW

File: lib/src/validator/language_version.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      if (unitLanguageVersion != null) {
+        if (Version(unitLanguageVersion.major, unitLanguageVersion.minor, 0) >
+            packageSdkMinVersion) {
+          final packageLanguageVersionString =
+              '${packageSdkMinVersion.major}.${packageSdkMinVersion.minor}';
+          final unitLanguageVersionString =
+              '${unitLanguageVersion.major}.${unitLanguageVersion.minor}';
+          final relativePath = p.relative(path);
+          errors.add('$relativePath is declaring language version '
+              '$unitLanguageVersionString that is newer than '
```
Comment (jonasfj) [inline lib/src/validator/language_version.dart]: > ..that is new than the SDK constraint "..." declared in `pubspec.yaml`

File: pubspec.yaml [hunk 2]
```diff
@@ -37,3 +37,15 @@ dev_dependencies:
   test: ^1.3.0
   test_descriptor: ^1.0.0
   test_process: ^1.0.0
+
+dependency_overrides:
```
Comment (jonasfj) [inline pubspec.yaml]: Please add a comment explaining the logic here..

File: lib/src/validator/language_version.dart (hunk 1 above)
Comment (sigurdm) [reply lib/src/validator/language_version.dart]: Done

File: pubspec.yaml (hunk 2 above)
Comment (sigurdm) [reply pubspec.yaml]: Done

**Pull request merged**